src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    for (const key in obj) {
+      if (obj[key] === 'addProperties') {
+        Object.assign(state, obj.extraData);
+      }
+
+      if (obj[key] === 'removeProperties') {
+        for (const props of obj.keysToRemove) {
+          delete state[props];
+        }
+      }
+
+      if (obj[key] === 'clear') {
+        for (const keys in state) {
+          delete state[keys];
+        }
+      }
```

Comment:
What if first condition turns out true? Computer would do 2 unnecessary checks after that. Switch/case would be a perfect fit here